Library/Homebrew/test/caveats_spec.rb

```diff
@@ -26,4 +26,65 @@ def caveats
       expect(described_class.new(f)).not_to be_empty
     end
   end
+
+  describe "#caveats" do
+    context "when f.plist is not nil" do
+      it "prints plist startup information when f.plist_startup is not nil" do
+        p = formula do
+              url "foo-1.0"
+              def plist
+                true
+              end
+              def plist_startup
+                true
+              end
+            end
+        expect(described_class.new(p).caveats).to include("startup:\n  sudo brew")
+      end
+
+      it "prints plist login information when f.plist_startup is nil" do
+        p = formula do
+              url "foo-1.0"
+              def plist
+                true
+              end
+            end
+        expect(described_class.new(p).caveats).to include("login:\n  brew")
+      end
+    end
+
+    context "when f.keg_only is not nil" do
+      it "gives information about command to be run when f.bin is a directory" do
+        p = formula do
+              url "foo-1.0"
+              def keg_only?
+                true
+              end
+              def bin
+                Pathname.new("bin")
+              end
+            end
+        p_inst = described_class.new(p)
+        allow(p.bin).to receive(:directory?).and_return(true)
+        expect(p_inst.caveats).to include("This formula is keg-only")
+        expect(p_inst.caveats).to include(Utils::Shell.prepend_path_in_profile(p.opt_bin.to_s))
+      end
+
+      it "gives information about command to be run when f.sbin is a directory" do
+        g = formula do
+              url "foo-1.0"
+              def keg_only?
+                true
+              end
+              def sbin
+                Pathname.new("sbin")
```

Comment:
Both `p.bin.directory?` and `g.sbin.directory?` give `false` even after mocking them (p and g are formulae).
If I mock values of `p.bin` and `g.sbin`, then the mocking for `p.bin.directory?` and `g.sbin.directory?` works.
Could you please help figure out why this is happening? @mistydemeo @MikeMcQuaid